x/sisu/client/rest/gas_handler.go

```diff
@@ -0,0 +1,78 @@
+package rest
+
+import (
+	"encoding/json"
+	"github.com/sisu-network/sisu/x/sisu/helper"
+	"math/big"
+	"net/http"
+
+	"github.com/sisu-network/lib/log"
+)
+
+type gasCostRequest struct {
+	Chain   string `json:"chain,omitempty"`
+	TokenId string `json:"token_id,omitempty"`
+}
+
+type gasCostResponse struct {
+	Chain   string `json:"chain,omitempty"`
+	TokenId string `json:"token_id,omitempty"`
+	GasCost int64  `json:"gas_cost,omitempty"`
+}
+
+func (a *ExternalHandler) newGasCostHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		req := &gasCostRequest{}
+		if err := ReadRESTReq(r, req); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		log.Info("Come here 1")
+		gasPrice, err := a.worldState.GetGasPrice(req.Chain)
+		if err != nil {
+			log.Error(err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		log.Info("Come here 2")
+		// TODO: correct gasLimit here
+		gasLimit := big.NewInt(8_000_000)
```

Comment:
This is the gasUnit, not gasLimit.

There is a lot of logic shared with the `callERC20TransferIn` in `tx_producer`. Can you move into both of them into 1 place. I think it might ok to put them into the WorldState